### PR TITLE
[SofaCUDA] Better readme

### PR DESCRIPTION
**Pull request by alxbilger**
To be merged after https://github.com/sofa-framework/sofa/pull/2588

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/SofaCUDA/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Performances
+
+The folder `scenes/benchmarks` contains simulation files used to compare performances between CPU and GPU implementations.
+All scenes exist in double: the CPU version has the suffix `_cpu`, and the GPU version has the suffix `_gpu`.
+
+To give you an idea of the speed up you can achieve using SofaCUDA, measurement have been performed on some systems on some simulations in `scenes/benchmarks`.
+It is provided as a CSV file in the `doc` folder.
+Here is the corresponding graph:
+![performances_csv](https://user-images.githubusercontent.com/21199245/148964875-99c21320-384f-40c9-9dc6-a5e77a8dca82.png)
```
Comment (epernod) [inline applications/plugins/SofaCUDA/README.md:37]: do you think it is better to put the graph directly inside the repo under /doc ?

File: applications/plugins/SofaCUDA/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+CUDA data types are defined in `CudaTypes.h`.
+
+In SOFA, components working with a specific data type are just template classes explicitly instantiated with this data type.
+For example, `MechanicalObject<Vec3Types>` is the instantiation of the template class `MechanicalObject` with `Vec3Types` as the template parameter.
+In SofaCUDA, components are template specializations of the SOFA equivalents.
+For example, `MechanicalObject<CudaVec3Types>` is a specialization of `MechanicalObject` for `CudaVec3Types`.
+
+The result is that the user just has to change the template parameter in the scene, in order to use the CUDA version of the same component.
+For example, in XML, `<MechanicalObject template="Vec3f" />` is the CPU version of `MechanicalObject`, and `<MechanicalObject template="CudaVec3f" />` is the GPU version.
+
```
Comment (epernod) [inline applications/plugins/SofaCUDA/README.md:28]:
```suggestion
Note that in SOFA Vec3Types are by default in double (Vec3d), except if you compile in float only (CMake option: SOFA_FLOATING_POINT_TYPE). However CudaVec3Types are using float on the device. You can activate double precision using CMake option: SOFACUDA_DOUBLE).

```

File: applications/plugins/SofaCUDA/README.md (hunk 1 above)
Comment (epernod) [reply applications/plugins/SofaCUDA/README.md:37]: @alxbilger ?

Comment (alxbilger) [reply applications/plugins/SofaCUDA/README.md:37]: I don't know if it's better, but I don't mind having it in this folder. What I would like, is something easy to read to realize the speedup compared to the CPU. For now, a graph image is a good solution.

**Pull request merged**